tests/unit/maze_transformer/generation/test_tokenizers.py

```diff
@@ -1,5 +1,276 @@
 """
 Test that the wrapped tokenizer loaded provides the same outputs
 as the original tokenizer (i.e. just using the token map in cfg)
+
+We may want a separate set of tests for different tokenization schemes
 """
-# TODO - Pending tokenizer
+import torch
+from pytest import mark, param
+from transformer_lens import HookedTransformer, HookedTransformerConfig
+
+from maze_transformer.training.config import ConfigHolder
+from maze_transformer.training.mazedataset import MazeDatasetConfig
+from maze_transformer.training.tokenizer import HuggingMazeTokenizer
+from scripts.create_dataset import generate_MazeTokenizer
+
+
+def test_tokenization_encoding():
+    # Check that wrapped tokenizer __call__ returns the same as original tokenizer
+    maze = generate_MazeTokenizer(None, 3, (0, 0), (2, 1))
+
+    # Need to generate a config to extract the token map >.<
+    cfg = MazeDatasetConfig(name="testing_maze", grid_n=3, n_mazes=1)
+    node_token_map = cfg.node_token_map
+
+    # Adjacency List Tokenization
+    maze_str_tokens = maze.as_tokens(node_token_map)
+
+    # Manual Tokenization
+    vocab_map = {k: v for v, k in enumerate(cfg.token_arr)}
+    maze_tokens = [vocab_map[token] for token in maze_str_tokens]
+
+    # WrappedTokenizer
+    # Initialized with a configholder - tokenizer will eventually be a string
+    cfg_holder = ConfigHolder(
+        train_cfg=None, dataset_cfg=cfg, model_cfg=None, tokenizer=None
+    )
+    tokenizer = HuggingMazeTokenizer(cfg_holder)
+
+    tokenizer_out = tokenizer(maze_str_tokens)["input_ids"]
+    assert torch.all(
+        torch.tensor(tokenizer_out).flatten() == torch.tensor(maze_tokens)
+    ), "Tokenization mismatch"
+
+
+def test_ascii_encoding():
+    # Check that the ascii encoding works for multiple different inputs
+    maze_str_tokens = [
+        "<ADJLIST_START>",
+        "(1,1)",
+        "<-->",
+        "(2,1)",
+        ";",
+        "(2,0)",
+        "<-->",
+        "(1,0)",
+        ";",
+        "(0,1)",
+        "<-->",
+        "(0,0)",
+        ";",
+        "(2,2)",
+        "<-->",
+        "(2,1)",
+        ";",
+        "(2,0)",
+        "<-->",
+        "(2,1)",
+        ";",
+        "(0,2)",
+        "<-->",
+        "(1,2)",
+        ";",
+        "(0,0)",
+        "<-->",
+        "(1,0)",
+        ";",
+        "(0,2)",
+        "<-->",
+        "(0,1)",
+        ";",
+        "<ADJLIST_END>",
+        "<TARGET_START>",
+        "(2,1)",
+        "<TARGET_END>",
+        "<START_PATH>",
```

Comment:
Overlap with [my PR](https://github.com/AISC-understanding-search/maze-transformer/pull/79/files#diff-23545957e6414d237a17e96a79f5d6a5db64b00d59859a50ce776e0a202ffa14), needs the new origin tokens